src/main/java/Duke.java

```diff
@@ -1,10 +1,126 @@
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    /**
+     * The constructor for the Duke class.
+     */
+    public Duke() {
+        ui = new Ui();
+        tasks = new TaskList();
+        storage = new Storage();
+        parser = new Parser();
+    }
+
+    /**
+     * The main method for Duke.
+     * @param args For the main method.
+     * @throws DukeException Throws DukeException.
+     * @throws IOException Throws IOException.
+     */
+    public static void main(String[] args) throws DukeException, IOException {
+        new Duke().run();
+    }
+
+
+    /**
+     * To run the duke program.
+     * @throws DukeException Throws DukeException.
+     * @throws IOException Throws IOException.
+     */
+    public void run() throws DukeException, IOException {
+        storage.loadData();
+        ui.welcomeMessage();
+        ui.reply("    What can I do for you");
+
+        Scanner sc = new Scanner(System.in);
+        String command;
+        while (!(command = sc.nextLine()).equals("bye")) {
+            try {
+                handle(command);
+            }
+            catch (DukeException ex) {
+                ui.showError(ex.getMessage());
+            }
+
+        }
+
+        storage.saveData();
+        ui.reply("    Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Handles the command provided by the user.
+     * @param command The user command.
+     * @throws DukeException Throws DukeException.
+     */
+    public void handle(String command) throws DukeException{
+        if (command.equals("list")) {
+            ui.printList();
+        } else {
+            String type = parser.getType(command);
+            if (type.equals("done") || type.equals("delete")) {
+                int taskNo = parser.getTaskIndex(command);
+                if (taskNo > tasks.numOfTasks() || taskNo <= 0) {
+                    throw new DukeException("☹ OOPS!! Not a valid number");
+                } else {
+                    if (type.equals("done")) {
```

Comment:
Try to avoid nested if conditions when you write code as they reduce readability. Think of possible ways to make it more readable e.g. switch case